nomt/src/cursor.rs

```diff
@@ -160,13 +160,14 @@ impl PageCacheCursor {
                 };
             }
             if self.depth as usize % DEPTH == 0 {
-                // attempt to load next page if we are at the end of our previous page or the root.
-                for _ in 0..PREFETCH_N {
-                    let page_id = match ppf.next() {
-                        Some(page) => page,
-                        None => break,
-                    };
-                    self.pages.prepopulate(page_id);
+                if self.depth as usize % PREFETCH_N == 0 {
```

Comment:
The intention is to fetch pages at every DEPTH * PREFETCH_N boundary?

if so, makes sense - I wonder why we haven't done this before. Aren't we loading a shit ton of unnecessary pages?